packages/drivers/routerlicious-driver/src/wholeSummaryDocumentStorageService.ts

```diff
@@ -275,33 +316,35 @@ export class WholeSummaryDocumentStorageService implements IDocumentStorageServi
 			},
 		);
 
-		assert(
-			normalizedWholeSummary.snapshotTree.id !== undefined,
-			0x275 /* "Root tree should contain the id" */,
-		);
-		const wholeFlatSummaryId: string = normalizedWholeSummary.snapshotTree.id;
-		const snapshotTreeVersion = {
-			id: wholeFlatSummaryId,
-			snapshotTree: normalizedWholeSummary.snapshotTree,
-		};
+		return normalizedWholeSummary;
+	}
 
+	private async initializeFromSnapshot(
+		normalizedWholeSummary: INormalizedWholeSummary,
+		versionId: string | null,
+	): Promise<string> {
+		const wholeFlatSummaryId = normalizedWholeSummary.snapshotTree.id;
+		assert(wholeFlatSummaryId !== undefined, 0x275 /* "Root tree should contain the id" */);
 		const cachePs: Promise<any>[] = [
-			this.snapshotTreeCache.put(this.getCacheKey(wholeFlatSummaryId), snapshotTreeVersion),
+			this.snapshotTreeCache.put(
+				this.getCacheKey(wholeFlatSummaryId),
+				normalizedWholeSummary,
+			),
 			this.initBlobCache(normalizedWholeSummary.blobs),
 		];
-		if (wholeFlatSummaryId !== versionId) {
+		if (wholeFlatSummaryId !== versionId && versionId !== null) {
 			// versionId could be "latest". When summarizer checks cache for "latest", we want it to be available.
 			// TODO: For in-memory cache, <latest,snapshotTree> will be a shared pointer with <snapshotId,snapshotTree>,
 			// However, for something like Redis, this will cache the same value twice. Alternatively, could we simply
 			// cache with versionId?
 			cachePs.push(
-				this.snapshotTreeCache.put(this.getCacheKey(versionId), snapshotTreeVersion),
```

Comment:
I'm not sure I follow (in general) why do we ever want to load back specific version of a file?
I believe we eliminated all such code paths in runtime. So saving latest (in a context of later - loading latest / the only version) should be enough.

Reply:
once we return the versions after fetching the latest snapshot in getVersions we supply back the versionIds which is used by loader to actually make the snapshot call with the particular version where it is will used. This is same as odsp where we have inmemory cache where we cache trees which will be used later. That is why we need to cache both.